File: website/src/main/java/info/xiaomo/website/service/ChangeLogService.java
package info.xiaomo.website.service;

import info.xiaomo.website.model.ChangeLogModel;
import org.springframework.data.domain.Page;

import java.util.List;

/**
 * 把今天最好的表现当作明天最新的起点．．～
 * いま 最高の表現 として 明日最新の始発．．～
 * Today the best performance  as tomorrow newest starter!
 * Created by IntelliJ IDEA.
 *
 * @author : xiaomo
 * github: https://github.com/xiaomoinfo
 * email: xiaomo@xiaomo.info
 * <p>
 * Date: 2016/4/1119:49
 * Copyright(©) 2015 by xiaomo.
 **/
public interface ChangeLogService {

    /**
     * 根据id查
     *
     * @param id
     * @return
     */
    ChangeLogModel findById(Long id);

    /**
     * 根据名字查
     *
     * @param name
     * @return
     */
    ChangeLogModel findByName(String name);

    /**
     * find all
     *
     * @param start
     * @param pageSize
     * @return
     */
    Page<ChangeLogModel> findAll(int start, int pageSize);

    /**
     * find all
     *
     * @return
     */
    List<ChangeLogModel> findAll();

    /**
     * add
     *
     * @param model
     * @return
     */
    ChangeLogModel add(ChangeLogModel model);

    /**
     * update
     *
     * @param model
     * @return
     */
    ChangeLogModel update(ChangeLogModel model);

    /**
     * delete
     *
     * @param id
     * @return
     */
    ChangeLogModel delete(Long id);
}

File: website/src/main/java/info/xiaomo/website/service/ChangeLogService.kt
package info.xiaomo.website.service

import info.xiaomo.website.model.ChangeLogModel
import org.springframework.data.domain.Page

/**
 * 把今天最好的表现当作明天最新的起点．．～
 * いま 最高の表現 として 明日最新の始発．．～
 * Today the best performance  as tomorrow newest starter!
 * Created by IntelliJ IDEA.
 *
 * @author : xiaomo
 * github: https://github.com/xiaomoinfo
 * email: xiaomo@xiaomo.info
 *
 * Date: 2016/4/1119:49
 * Copyright(©) 2015 by xiaomo.
 */
interface ChangeLogService {

    /**
     * 根据id查
     *
     * @param id
     * @return
     */
    fun findById(id: Long?): ChangeLogModel

    /**
     * 根据名字查
     * @param name
     * @return
     */
    fun findByName(name: String): ChangeLogModel

    /**
     * find all
     * @param start
     * @param pageSize
     * @return
     */
    fun findAll(start: Int, pageSize: Int): Page<ChangeLogModel>

    /**
     * find all
     * @return
     */
    fun findAll(): List<ChangeLogModel>

    /**
     * add
     * @param model
     * @return
     */
    fun add(model: ChangeLogModel): ChangeLogModel

    /**
     * update
     * @param model
     * @return
     */
    fun update(model: ChangeLogModel): ChangeLogModel

    /**
     * delete
     * @param id
     * @return
     */
    fun delete(id: Long?): ChangeLogModel
}
